src/hooks/UseDynamicHighilghtedLink.js

```diff
@@ -0,0 +1,140 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+/**
+ * Custom hook to extract the logic behind the highlighted link in the AutoScrollableMenu
+ *
+ * This hook allow to determine which link should be highlighted based on two inputs.
+ *
+ *  - The first input is the menu link on which we clicked when we click on one.
```

Comment:
```suggestion
The first input is the menu link on which we clicked when we click on one link.
```
Not sure if that's what you meant.

Reply:
It is not exactly what I meant no, but I agree that it is a bit confusing. I was referring to the `two inputs` as being the `two input` to the "algorithm" that determine which of the link is currently highlighted, not the `two inputs` of the hook itself.

But I will change the comment to make it more understandable. Thanks for pointing it out.